src/components/InterestForm/index.tsx

```diff
@@ -2,72 +2,111 @@
-  RadioGroup,
-  Radio,
-  RadioInput,
   FormFooter,
   FormWarning,
   ErrorText,
-  RadioLabel,
 } from './styles';
 
-const radioOptions = [
-  'Attorney',
-  'Interpreter',
-  'Either Attorney or Interpreter',
-];
+const formQuestion = (label: string, required = false) => (
+  <Flex $direction="row">
+    <FormQuestion>{label}</FormQuestion>
+    {required ? (
+      <FormQuestion $color={COLORS.redMid}>&nbsp;*</FormQuestion>
+    ) : (
+      <FormQuestion>&nbsp;(optional)</FormQuestion>
+    )}
+  </Flex>
+);
 
-export default function InterestForm({ caseData }: { caseData: CaseListing }) {
+export default function InterestForm({
+  listingData,
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
   const auth = useAuth();
   const [reason, setReason] = useState<string>('');
-  const [rolesInterested, setRolesInterested] = useState<string>('');
+  const [needsInterpreter, setNeedsInterpreter] = useState<string>('');
   const [startDate, setStartDate] = useState<string>('');
   const [submitted, setSubmitted] = useState(false);
   const [missingInfo, setMissingInfo] = useState(false);
 
   useEffect(() => {
     // Reset form fields when caseData changes
     setReason('');
+    setNeedsInterpreter('');
     setStartDate('');
-    setRolesInterested('');
     setSubmitted(false);
     setMissingInfo(false);
-  }, [caseData]);
+  }, [listingData]);
 
   const handleInsert = async () => {
-    if (startDate === '' || rolesInterested === '') {
+    // Error handling
+    if (
+      (listingData.listing_type === 'CASE' &&
+        !interpretation &&
+        (startDate === '' || needsInterpreter === '')) ||
+      (((listingData.listing_type === 'CASE' && interpretation) ||
+        listingData.listing_type === 'INT') &&
+        startDate === '') ||
+      ((listingData.listing_type === 'LCA' ||
+        listingData.listing_type === 'DOC') &&
+        reason === '')
+    ) {
       setMissingInfo(true);
       return;
     }
-    if (isValidDate(startDate)) {
+    let responses = {};
+    if (
+      isValidDate(startDate) ||
+      listingData.listing_type === 'DOC' ||
+      listingData.listing_type === 'LCA'
+    ) {
       if (auth && auth.userId) {
+        if (
+          listingData.listing_type === 'CASE' ||
+          listingData.listing_type === 'INT'
+        ) {
+          responses = { ...responses, start_date: new Date(startDate) };
+          if (listingData.listing_type === 'CASE' && !interpretation) {
+            responses = {
+              ...responses,
+              needs_interpreter: needsInterpreter === 'Yes',
+            };
+          }
+        }
+        responses = { ...responses, interest_reason: reason };
         const newInterest: Interest = {
-          listing_id: caseData.id,
-          listing_type: 'CASE',
+          listing_id: listingData.id,
+          listing_type:
+            listingData.listing_type === 'CASE' && interpretation
+              ? 'CASE_INT'
```

Comment:
It seems like even when I submit an interest just for a regular case (as an attorney), the listing type of the interest submission is `CASE_INT`? Is that expected behavior?